java-lib/src/main/java/com/yammer/metrics/core/DeltaCounter.java

```diff
@@ -2,39 +2,83 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.wavefront.common.MetricConstants;
+import com.yammer.metrics.Metrics;
 
 /**
  * A counter for Wavefront delta metrics.
  *
- * Differs from a counter in that it is reset in the SocketMetricsProcessor every time the value is
- * reported.
+ * Differs from a counter in that it is reset every time the value is reported.
  *
- * (This is similar to how WavefrontHistogram is implemented)
+ * (This is similar to how {@link com.yammer.metrics.core.WavefrontHistogram} is implemented)
  *
- * @author Suranjan Pramanik (spramanik@wavefront.com)
+ * @author Suranjan Pramanik (suranjan@wavefront.com)
  */
 public class DeltaCounter extends Counter {
 
-    private DeltaCounter() {
-        // nothing to do. keeping it private so that it can't be instantiated directly
+  private DeltaCounter() {
+    // nothing to do. keeping it private so that it can't be instantiated directly
+  }
+
+  @VisibleForTesting
+  public static synchronized DeltaCounter get(MetricsRegistry registry, MetricName metricName) {
+    if (registry == null || metricName == null || metricName.getName().isEmpty()) {
+      throw new IllegalArgumentException("Invalid arguments");
+    }
+
+    DeltaCounter counter = new DeltaCounter();
+    MetricName newMetricName = getDeltaCounterMetricName(metricName);
+    registry.getOrAdd(newMetricName, counter);
+    return counter;
+  }
+
+  public static DeltaCounter get(MetricName metricName) {
+    return get(Metrics.defaultRegistry(), metricName);
+  }
+
+  public static long getCount(Counter counter) {
```

Comment:
I think the name of this method makes it looks benign and immutable but it has a serious side effect. as soon as you invoke getCount() it will decrement the value to 0
maybe rename it to processDeltaCounter and also add a javadoc stating what happens when you invoke this method.

Reply:
Need javadoc then for sure